pkg/bgpv1/manager/reconcilerv2/policies_test.go

```diff
@@ -483,8 +483,21 @@ func Test_MergeRoutePolicies(t *testing.T) {
 				req.NoError(err)
 			}
 
-			req.Equal(tt.expected, result)
+			EqualRoutePolicyUnorderedStatements(req, tt.expected, result)
 		})
 	}
+}
+
+func EqualRoutePolicyUnorderedStatements(req *require.Assertions, policyA, policyB *types.RoutePolicy) {
+	if policyA == nil && policyB == nil {
+		return
+	}
+	if policyA == nil || policyB == nil {
+		req.Fail("policies are not equal")
+		return
+	}
 
+	req.Equal(policyA.Name, policyB.Name)
+	req.Equal(policyA.Type, policyB.Type)
+	req.ElementsMatch(policyA.Statements, policyB.Statements)
```

Comment:
I'd slightly prefer sorting statements first and compare the entire struct using `Equal` instead of implementing our own `Equal` here, so that we don't need to change this function when we add a new field.